### ISSUE TITLE
Run multimer with a "single receptor" and "multiple partners"

### ISSUE THREAD

**Issue by Cucco23** [google-deepmind/alphafold#902]
Hallo all,

Let's say that I have a protein to which I would like to dock multiple different proteins but always in a 1:1 ratio.
there is a way to provide a list in the fasta file so that the first protein (receptor) does not require to be re-calculated?
e.g. protein A with proteins B, C, D, E....

I hope my question was clear and that someone has a good suggestion

Thank you!

**Comment by tejhande** [comment]
Certainly! You can dock multiple ligands (proteins B, C, D, E, etc.) to a fixed receptor (protein A) using molecular docking software that supports docking multiple ligands simultaneously in a single run. Prepare your receptor and ligands, create input files specifying them, and run the docking simulation. Analyze the results afterward. This approach saves time and resources compared to docking each ligand individually.

import subprocess

def run_autodock_vina(receptor_file, ligand_files, output_dir):
    for ligand_file in ligand_files:
        output_file = f"{output_dir}/output_{ligand_file.split('.')[0]}.pdbqt"
        command = f"vina --receptor {receptor_file} --ligand {ligand_file} --out {output_file} --log {output_dir}/log.txt"
        subprocess.run(command, shell=True)

if __name__ == "__main__":
    receptor_file = "receptor.pdbqt"
    ligand_files = ["ligand1.pdbqt", "ligand2.pdbqt", "ligand3.pdbqt"]
    output_dir = "dock_output"

    run_autodock_vina(receptor_file, ligand_files, output_dir)

**Comment by Cucco23** [comment]
Thank you for the answer.

I meant by using alphafold multimer.

Moreover, are you sure you can dock proteins with vina? Where is this publushed? To my knowledge it can be used only with small molecules since it test conformations.

On another note lets say af multimer provides me a model. How do you test the quality of the docking?

Thanks

________________________________

**Comment by genomicsgroup** [comment]
The Alphafold source code does not support this feature, but my solution is manually generates MSA and imports it into the corresponding output folder.  The simplest solution is to first calculate A-B, copy their msas folder to the output folder of A-C, but please note that it is important to ensure that the protein A is in the first position in the fasta file and delete the B folder inside the msas folder. This will automatically read the MSA of protein A and calculate the MSA of protein C when you start with --use_precomputed_msas flag. @Cucco23

**Comment by Cucco23** [comment]
Thank you very much for the suggestion!
I will try this.

**Comment by tcoates5** [comment]
I believe the AlphaPulldown repository (by the Kosinski lab) seems to match what you want to use AF to do
